flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/hybrid/tiered/storage/SortBufferAccumulator.java

```diff
@@ -94,29 +99,58 @@ public class SortBufferAccumulator implements BufferAccumulator {
     @Nullable
     private TriConsumer<TieredStorageSubpartitionId, Buffer, Integer> accumulatedBufferFlusher;
 
+    /**
+     * An executor to periodically check the size of buffer pool. If the size is changed, the
+     * accumulated buffers should be flushed to release the buffers.
+     */
+    private final ScheduledExecutorService periodicalAccumulatorFlusher =
+            Executors.newSingleThreadScheduledExecutor(
```

Comment:
If we register a listener for each LBP, when a redistribution happens, all listeners need to do some actions(write operations) and these actions need to acquire their own locks. So we have to coordinate the lock `availableMemorySegments` in `NetworkBufferPool` and all other locks. I tried listen-and-notify strategy at first but found that it is very easy to cause a deadlock. 
I agree that a listen-and-notify strategy is better if we can acquire the lock `availableMemorySegments` in each listener to avoid the deadlock, but exposing that lock object should be careful and I think it is out of scope.